### ISSUE TITLE
SCSS line comments

### ISSUE THREAD

**Issue by borodean** [stylelint/stylelint#519]
[postcss-scss](https://github.com/postcss/postcss-scss) interprets line commets as they were block ones. This causes line comments always fail on `comment-space-inside: "always"` rule:

```scss
// stylelint thinks this comment requires single spaces at both sides
```

And `comment-empty-line-before: "always"` rule:

```scss
// stylelint considers each of those lines a separate comment,
// so it expects empty lines between them.
```

Apart from that, you may want separate rules for line and block comments.

cc @ai @ben-eb

**Comment by ai** [comment]
@borodean I am not maintainer for Stylelint. You should ask @davidtheclark.

**Comment by borodean** [comment]
@ai I'm notifying you because this issue relies on how postcss-scss works and may require improvements in it.

**Comment by davidtheclark** [comment]
Tough one. Best solution that comes to mind initially is to have the the `comment-space-inside` rule ignore `//` comments altogether. It makes sense to me that those comment nodes end up with different metadata and so confuse things. And `comment-empty-line-before` could have an `ignore: [ 'comment' ]` option. What do you think?

**Comment by borodean** [comment]
@davidtheclark what do you mean by different metadata? Also, how do you expect `ignore: [ 'comment' ]` thingie to work?

**Comment by davidtheclark** [comment]
@borodean I guess it was an awkward way of summarizing what you pointed out: the nodes have different properties (spacing, line breaking) than you would expect from a block comment.

**Comment by davidtheclark** [comment]
> Also, how do you expect ignore: [ 'comment' ] thingie to work?

It would mean that stylelint would not complain if there are two comments in a row without an empty line between. This would allow you to use successive `//` comments without problems. It would would still expect empty lines before comments if the previous node is not another comment.

**Comment by jeddy3** [comment]
> comment-space-inside rule ignore // comments altogether

Sgtm.

>  And comment-empty-line-before could have an ignore: [ 'comment' ] option

Likewise.

@borodean As in #518, feel free to create a plugin to require or disallow a space after the `//` e.g. something like `stylelint-double-slash-comment-space-after: "always"|"never"`, or whatever you'd like to call it :)

**Comment by borodean** [comment]
@davidtheclark @jeddy3 why don't you also make the `comment-empty-line-before` ignore `//`?

My five cents: I think that the logic behind fixing comment-related rules should not be ignoring `//`, but accepting only `/* */`, beacuse who knows what kind of comments could be introduced in preprocessors in the future.

As for the `ignore: [ 'comment' ]` thingie, I don't understand why there needs to be a special case to designate subsequent block comments, if non-CSS line comments would be skipped for all comment-related rules.

**Comment by davidtheclark** [comment]
@borodean I agree with you.

**Comment by s10wen** [comment]
Am I right in thinking adding:
```
"comment-empty-line-before": "never"
```
to my `.stylelintrc` file, I should no longer have `//` comment errors? (I've tried this, but I'm still getting them)

**Comment by jeddy3** [comment]
@s10wen Are you using the `scss` syntax option? Look [here](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/css-processors.md#parsing-scss) if your using the CLI or Node API, and [at the 2nd usage example here](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/postcss-plugin.md#usage-examples) if you're using the PostCSS plugin.

**Comment by s10wen** [comment]
@jeddy3 thanks for the reply, I'm using the CLI, is there a `scss` syntax option for that please?

**Comment by davidtheclark** [comment]
@s10wen : https://github.com/stylelint/stylelint/blob/master/docs/user-guide/css-processors.md#parsing-scss

**Comment by s10wen** [comment]
@davidtheclark awesome, thanks, sorry I missed that bit, I've just started playing around it, but looking great so far :+1:
